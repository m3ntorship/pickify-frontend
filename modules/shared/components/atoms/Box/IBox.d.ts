import type { ReactElement } from 'react';

declare namespace IBox {
  export interface IProps {
    isGreyColor?: boolean;
    isWhiteColor?: boolean;
    children: ReactElement;
    withDevider?: boolean;
  }
}

export { IBox };
