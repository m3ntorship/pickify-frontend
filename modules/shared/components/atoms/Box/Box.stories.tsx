import React from 'react';
import type { Story, Meta } from '@storybook/react';
import Box from './Box';
import type { IBox } from './IBox';

export default {
  title: 'Atoms/Box',
  component: Box,
} as Meta;

const Template: Story<IBox.IProps> = (args) => (
  <div>
    <div className="mb-12">
      <h1>With Header</h1>
      <Box {...args}>
        <>
          <Box.Header withDevider>
            <>
              <div>left</div>
              <div>right</div>
            </>
          </Box.Header>
          <Box.Body>
            <p>
              Lorem Ipsum is simply dummy text of the printing and typesetting
              industry. Lorem Ipsum has been the industrys standard dummy text
              ever since the 1500s,
            </p>
          </Box.Body>
        </>
      </Box>
    </div>
    <div className="mb-12">
      <h1>With Footer</h1>
      <Box {...args}>
        <>
          <Box.Body>
            <p>
              Lorem Ipsum is simply dummy text of the printing and typesetting
              industry. Lorem Ipsum has been the industrys standard dummy text
              ever since the 1500s,
            </p>
          </Box.Body>
          <Box.Footer withDevider>
            <>
              <div>left</div>
              <div>right</div>
            </>
          </Box.Footer>
        </>
      </Box>
    </div>
    <div className="mb-12">
      <h1>With Header & Footer</h1>
      <Box {...args}>
        <>
          <Box.Header withDevider>
            <>
              <div>left</div>
              <div>right</div>
            </>
          </Box.Header>
          <Box.Body>
            <p>
              Lorem Ipsum is simply dummy text of the printing and typesetting
              industry. Lorem Ipsum has been the industrys standard dummy text
              ever since the 1500s,
            </p>
          </Box.Body>
          <Box.Footer withDevider>
            <>
              <div>left</div>
              <div>right</div>
            </>
          </Box.Footer>
        </>
      </Box>
    </div>
  </div>
);

export const Default = Template.bind({});
Default.args = {
  isGreyColor: true,
  isWhiteColor: false,
  classes: '',
};
