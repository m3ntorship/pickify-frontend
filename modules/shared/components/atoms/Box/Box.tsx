import React from 'react';
import type { ReactElement } from 'react';
import classNames from 'classnames';
import styles from './Box.module.css';
import type { IBox } from './IBox';
import Divider from '../Divider/Divider';
import * as EDivider from '../Divider/types/EDivider';

const Box = ({
  classes,
  isGreyColor,
  isWhiteColor,
  children,
}: IBox.IProps): ReactElement => {
  const boxClasses = classNames(
    styles.box,
    {
      'bg-grey-bg': isGreyColor,
      'bg-white': isWhiteColor,
    },
    classes,
  );
  return <div className={boxClasses}>{children}</div>;
};

const BoxDevider = (): ReactElement => {
  return <Divider type={EDivider.DividerType.Horizontal} length="100%" />;
};

const Header = ({
  classes,
  children,
  withDevider,
}: IBox.IProps): ReactElement => {
  const headerClasses = classNames(styles['header-wrapper'], classes);
  return (
    <div className={headerClasses}>
      <div className={styles.header}>{children}</div>
      {withDevider && (
        <div className="mt-3">
          <BoxDevider />
        </div>
      )}
    </div>
  );
};

const Body = ({ classes, children }: IBox.IProps): ReactElement => {
  const bodyClasses = classNames(classes);
  return <div className={bodyClasses}>{children}</div>;
};

const Footer = ({
  classes,
  children,
  withDevider,
}: IBox.IProps): ReactElement => {
  const footerClasses = classNames(styles['footer-wrapper'], classes);
  return (
    <div className={footerClasses}>
      {withDevider && (
        <div className="mb-4">
          <BoxDevider />
        </div>
      )}
      <div className={styles.footer}>{children}</div>
    </div>
  );
};

Box.Header = Header;
Box.Body = Body;
Box.Footer = Footer;

export default Box;
