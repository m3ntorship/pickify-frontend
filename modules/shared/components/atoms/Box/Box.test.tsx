import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Box from './Box';

describe('Friend Suggestion Component', () => {
  it('Should render Header, Body and Footer with Divider grey background', () => {
    const tree = renderer
      .create(
        <Box isGreyColor>
          <div>
            <Box.Header withDevider>
              <p>Dummy Header</p>
            </Box.Header>
            <Box.Body>
              <p>Dummy Body</p>
            </Box.Body>
            <Box.Footer withDevider>
              <p>Dummy Footer</p>
            </Box.Footer>
          </div>
        </Box>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should render Header, Body and Footer with Divider white background', () => {
    const tree = renderer
      .create(
        <Box isWhiteColor>
          <div>
            <Box.Header withDevider>
              <p>Dummy Header</p>
            </Box.Header>
            <Box.Body>
              <p>Dummy Body</p>
            </Box.Body>
            <Box.Footer withDevider>
              <p>Dummy Footer</p>
            </Box.Footer>
          </div>
        </Box>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should render Header, Body and Footer', () => {
    const tree = renderer
      .create(
        <Box isWhiteColor>
          <div>
            <Box.Header>
              <p>Dummy Header</p>
            </Box.Header>
            <Box.Body>
              <p>Dummy Body</p>
            </Box.Body>
            <Box.Footer>
              <p>Dummy Footer</p>
            </Box.Footer>
          </div>
        </Box>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render Header', () => {
    const tree = renderer
      .create(
        <Box isWhiteColor>
          <Box.Header>
            <p>Dummy Header</p>
          </Box.Header>
        </Box>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render Body', () => {
    const tree = renderer
      .create(
        <Box isWhiteColor>
          <Box.Body>
            <p>Dummy Body</p>
          </Box.Body>
        </Box>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render Footer', () => {
    const tree = renderer
      .create(
        <Box isWhiteColor>
          <Box.Footer>
            <p>Dummy Footer</p>
          </Box.Footer>
        </Box>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
