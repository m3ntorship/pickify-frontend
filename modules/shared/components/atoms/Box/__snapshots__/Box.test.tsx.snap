// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Friend Suggestion Component Should render Body 1`] = `
<div
  className="box bg-white"
>
  <div>
    <p>
      Dummy Body
    </p>
  </div>
</div>
`;

exports[`Friend Suggestion Component Should render Footer 1`] = `
<div
  className="box bg-white"
>
  <div
    className="footer-wrapper"
  >
    <div
      className="footer"
    >
      <p>
        Dummy Footer
      </p>
    </div>
  </div>
</div>
`;

exports[`Friend Suggestion Component Should render Header 1`] = `
<div
  className="box bg-white"
>
  <div
    className="header-wrapper"
  >
    <div
      className="header"
    >
      <p>
        Dummy Header
      </p>
    </div>
  </div>
</div>
`;

exports[`Friend Suggestion Component Should render Header, Body and Footer 1`] = `
<div
  className="box bg-white"
>
  <div>
    <div
      className="header-wrapper"
    >
      <div
        className="header"
      >
        <p>
          Dummy Header
        </p>
      </div>
    </div>
    <div>
      <p>
        Dummy Body
      </p>
    </div>
    <div
      className="footer-wrapper"
    >
      <div
        className="footer"
      >
        <p>
          Dummy Footer
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Friend Suggestion Component Should render Header, Body and Footer with Divider grey background 1`] = `
<div
  className="box bg-grey-bg"
>
  <div>
    <div
      className="header-wrapper"
    >
      <div
        className="header"
      >
        <p>
          Dummy Header
        </p>
      </div>
      <div
        className="mt-3"
      >
        <div
          className="bg-grey-shd6 h-px"
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <div>
      <p>
        Dummy Body
      </p>
    </div>
    <div
      className="footer-wrapper"
    >
      <div
        className="mb-4"
      >
        <div
          className="bg-grey-shd6 h-px"
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="footer"
      >
        <p>
          Dummy Footer
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Friend Suggestion Component Should render Header, Body and Footer with Divider white background 1`] = `
<div
  className="box bg-white"
>
  <div>
    <div
      className="header-wrapper"
    >
      <div
        className="header"
      >
        <p>
          Dummy Header
        </p>
      </div>
      <div
        className="mt-3"
      >
        <div
          className="bg-grey-shd6 h-px"
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <div>
      <p>
        Dummy Body
      </p>
    </div>
    <div
      className="footer-wrapper"
    >
      <div
        className="mb-4"
      >
        <div
          className="bg-grey-shd6 h-px"
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="footer"
      >
        <p>
          Dummy Footer
        </p>
      </div>
    </div>
  </div>
</div>
`;
